app/main.py

```
@@ -1,3 +1,61 @@
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: object) -> "Distance":
+        if isinstance(other, Distance):
+            total_km = self.km + other.km
+        elif isinstance(other, (int, float)):
+            total_km = self.km + other
+        return Distance(total_km)
+
+    def __iadd__(self, other: object) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        return self
```

instead of returning self with no changes, you can raise TypeError